parsl/executors/globus_compute.py

```diff
@@ -0,0 +1,171 @@
+        super().__init__()
+        self.endpoint_id = endpoint_id
+        self.task_group_id = task_group_id
+        self.resource_specification = resource_specification
+        self.user_endpoint_config = user_endpoint_config
+        self.label = label
+        self.batch_size = batch_size
+        self.amqp_port = amqp_port
+        self.client = client
+        self.executor_kwargs = kwargs
+
+        if not _globus_compute_enabled:
+            raise OptionalModuleMissing(
+                ['globus-compute-sdk'],
+                "GlobusComputeExecutor requires globus-compute-sdk installed"
+            )
+
+    def start(self) -> None:
+        """ Start the Globus Compute Executor """
+
+        self._executor: Executor = Executor(
+            endpoint_id=self.endpoint_id,
+            task_group_id=self.task_group_id,
+            resource_specification=self.resource_specification,
+            user_endpoint_config=self.user_endpoint_config,
+            label=self.label,
+            batch_size=self.batch_size,
+            amqp_port=self.amqp_port,
+            client=self.client,
+            **self.executor_kwargs
+        )
+
+    def submit(self, func: Callable, resource_specification: Dict[str, Any], *args: Any, **kwargs: Any) -> Future:
+        """ Submit func to globus-compute
+
+
+        Parameters
+        ----------
+
+        func: Callable
+            Python function to execute remotely
+
+        resource_specification: Dict[str, Any]
+            Resource specification can be used specify MPI resources required by MPI applications on
+            Endpoints configured to use globus compute's MPIEngine. GCE also accepts *user_endpoint_config*
+            to configure endpoints when the endpoint is a `Multi-User Endpoint
+            <https://globus-compute.readthedocs.io/en/latest/endpoints/endpoints.html#templating-endpoint-configuration>`_
+
+        args:
+            Args to pass to the function
+
+        kwargs:
+            kwargs to pass to the function
+
+        Returns
+        -------
+
+        Future
+        """
+        res_spec = copy.deepcopy(resource_specification or self.resource_specification)
+        # Pop user_endpoint_config since it is illegal in resource_spec for globus_compute
+        if res_spec:
+            user_endpoint_config = res_spec.pop('user_endpoint_config', self.user_endpoint_config)
+        else:
+            user_endpoint_config = self.user_endpoint_config
+
+        self._executor.resource_specification = res_spec
+        self._executor.user_endpoint_config = user_endpoint_config
+        return self._executor.submit(func, *args, **kwargs)
+
+    def shutdown(self):
+        """Clean-up the resources associated with the Executor.
+
+        GCE.shutdown will cancel all futures that have not yet registered with
+        Globus Compute and will not wait for the launched futures to complete.
+        """
+        self._executor.shutdown(wait=False, cancel_futures=True)
```

Comment:
If a user (or Parsl developer) mistakenly calls `.shutdown()` on a GCE instance that hasn't been started, this currently explodes.  What about being more friendly with a simple check:

```python
if not self._executor:
    raise RuntimeError("Executor not started.")
```

That's just a smidgen less cryptic than `None has no attribute .shutdown`

The same suggestion goes for `.submit()`

Reply:
Fixed now as `self.executor` is set at init.